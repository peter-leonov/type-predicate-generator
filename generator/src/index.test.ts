import { expect, test } from "vitest";
import { generateForPlayground } from "./index";

test(generateForPlayground, () => {
  expect(
    generateForPlayground("export type X = number", "./foo")
  ).toMatchSnapshot();
});
