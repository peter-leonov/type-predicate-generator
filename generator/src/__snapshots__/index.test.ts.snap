// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`generateForPlayground 1`] = `
{
  "predicatesCode": "import { type X } from "./types";
export function isX(root: unknown): root is X {
    if (!(typeof root === "number")) {
        return false;
    }
    (root) satisfies X;
    return true;
}
",
  "testsCode": "import { expect, describe, it } from "./vitest";
import { isX } from "./types_guards";
const invalidValue: any = Symbol("invalidValue");
const valid_X = [
    0,
    42
];
const invalid_X = [
    invalidValue
];
describe("X", () => {
    it.for(valid_X)("valid", (value: unknown) => {
        expect(isX(value)).toBe(true);
    });
    it.for(invalid_X)("invalid", (value: unknown) => {
        expect(isX(value)).toBe(false);
    });
});
",
}
`;
